### PR TITLE
Identity work around for issue with non GUID Client Ids

### LINKED ISSUE #14585: [BUG] Identity SDK in source expecting Client-ID to always be a GUID
**Describe the bug**
Identity SDK in the source/master is expecting the Client-ID parameter to always be a GUID.

**Expected behavior**
What is the expected behavior?
It should work for non GUID client Ids too.

**Actual behavior (include Exception or Stack Trace)**
What is the actual behavior?

Throws the following exception:
Azure.Identity.AuthenticationFailedException: InteractiveBrowserCredential authentication failed: Error: ClientId is not a Guid.
 ---> MSAL.NetCore.4.16.1.0.MsalClientException:
        ErrorCode: client_id_must_be_guid
Microsoft.Identity.Client.MsalClientException: Error: ClientId is not a Guid.
   at Microsoft.Identity.Client.AbstractApplicationBuilder`1.Validate()
   at Microsoft.Identity.Client.PublicClientApplicationBuilder.Validate()
   at Microsoft.Identity.Client.Extensions.Msal.MsalCacheHelper.GetAccountIdentifiersAsync(StorageCreationProperties storageCreationProperties, TraceSourceLogger logger)
   at Microsoft.Identity.Client.Extensions.Msal.MsalCacheHelper.CreateAsync(StorageCreationProperties storageCreationProperties, TraceSource logger)
   at Azure.Identity.MsalClientBase`1.CreateCacheHelper(StorageCreationProperties storageProperties, Boolean async)
   at Azure.Identity.MsalClientBase`1.GetClientAsync(Boolean async, CancellationToken cancellationToken)
   at Azure.Identity.MsalPublicClient.AcquireTokenInteractiveAsync(String[] scopes, Prompt prompt, Boolean async, CancellationToken cancellationToken)
   at Azure.Identity.InteractiveBrowserCredential.GetTokenViaBrowserLoginAsync(String[] scopes, Boolean async, CancellationToken cancellationToken)
   at Azure.Identity.InteractiveBrowserCredential.GetTokenImplAsync(Boolean async, TokenRequestContext requestContext, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Azure.Identity.CredentialDiagnosticScope.FailWrapAndThrow(Exception ex)
   at Azure.Identity.InteractiveBrowserCredential.GetTokenImplAsync(Boolean async, TokenRequestContext requestContext, CancellationToken cancellationToken)
   at Azure.Core.Pipeline.TaskExtensions.EnsureCompleted[T](ValueTask`1 task)
   at Azure.Identity.InteractiveBrowserCredential.GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
   at ConsoleApp1.Program.Main(String[] args) in C:\Users\uxjava\source\repos\ConsoleApp1\ConsoleApp1\Program.cs:line 27

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1.Run the following code using the .NET Identity SDK in source: 

```net
            InteractiveBrowserCredentialOptions ibcOptions = new InteractiveBrowserCredentialOptions();
            ibcOptions.ClientId = "PublicClientId";
            ibcOptions.TenantId = "adfs";
            ibcOptions.AuthorityHost = new Uri("https://fs.msidlab8.com/");
            ibcOptions.EnablePersistentCache = true;

            InteractiveBrowserCredential ibc = new InteractiveBrowserCredential(ibcOptions);
            string[] bc = new string[1];
            bc[0] = "openid";

            try
            {
                string token = ibc.GetToken(new Azure.Core.TokenRequestContext(bc)).Token;
            } catch (Exception e)
            {
                Console.WriteLine(e.ToString());
            }
```

**Environment:**
 - Identity SDK: 1.3.0-dev
 -  Windows 10
- IDE and version : Visual Studio

### PR DESCRIPTION
**Pull request by schaabs**
This is a work around for issue https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/issues/98 in the .NET Msal Extension library. Currently constructing the `MsalCacheHelper` from this library when a non-GUID client id is used. This work around uses a random GUID for the Client Id to work around this issue. The random Client Id does not affect authentication in any way. MsalCacheHelper only uses the Client to create a `PublicClientApplication`  to enumerate accounts in the cache.

Fixes #14585

### CODE REVIEW

**Comment by AlexanderSher** [comment]
Is it possible to write a test for it, or we don't have proper environment yet?

File: sdk/identity/Azure.Identity/src/MsalClientBase.cs [hunk 1]
```diff
@@ -60,9 +60,13 @@ protected async ValueTask<TClient> GetClientAsync(bool async, CancellationToken
 
             if (EnablePersistentCache)
             {
+                // we are creating the MsalCacheHelper with a random guid based clientId to work around issue https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/issues/98
+                // This does not impact the functionality of the cacheHelper as the ClientId is only used to iterate accounts in the cache not for authentication purposes.
+                string msalCacheClientId = Guid.NewGuid().ToString();
```
Comment (AlexanderSher) [inline sdk/identity/Azure.Identity/src/MsalClientBase.cs:65]: Can we cache it as a static field?

Comment (schaabs) [reply sdk/identity/Azure.Identity/src/MsalClientBase.cs]: good suggestion, thanks!

**Comment by schaabs** [comment]
@AlexanderSher the plan is to add live and recorded tests for ADFS authentication for all credentials but we're still working out the environment details. We have manually verified the fix, and I'd like to merge it to unblock progress in this area.

**Review by AlexanderSher — APPROVED** [review]

**Pull request merged**